AFQ/api.py

```diff
@@ -854,12 +861,43 @@ def _wm_mask(self, row):
             if 'seg_file' in row.index and isinstance(self.wm_criterion, list):
                 wm_mask, meta = self._mask_from_seg(row, self.wm_criterion)
             else:
-                # Otherwise, we'll identify the white matter based on FA:
-                fa_fname = self._dti_fa(row)
-                dti_fa = nib.load(fa_fname).get_fdata()
-                wm_mask = dti_fa > self.wm_criterion
-                meta = dict(source=fa_fname,
-                            fa_threshold=self.wm_criterion)
+                # Otherwise, we'll identify the white matter based on scalars:
+                valid_scalars = list(self._scalar_dict.keys())
+                wm_mask = None
+                bounds = []
+                for bound, constraint in self.wm_criterion.items():
+                    for scalar, threshold in constraint.items():
+                        if scalar not in valid_scalars:
+                            raise RuntimeError((
+                                f"wm_criterion scalars should be one of"
+                                f" {', '.join(valid_scalars)}"))
+
+                        scalar_fname = self._scalar_dict[scalar](self, row)
+                        if bound == "lb":
+                            new_wm_mask = \
+                                nib.load(
+                                    scalar_fname).get_fdata() > threshold
+                            for_meta_bound_type = " > "
+                        elif bound == "ub":
+                            new_wm_mask = \
+                                nib.load(
+                                    scalar_fname).get_fdata() < threshold
+                            for_meta_bound_type = " < "
+                        else:
+                            raise RuntimeError("wm_criterion dictionary "
+                                               + " formatted incorrectly. See"
+                                               + " the default for reference")
+
+                        if wm_mask is None:
+                            wm_mask = new_wm_mask
+                        else:
+                            wm_mask = np.logical_and(wm_mask, new_wm_mask)
+                        bounds.append(
+                            scalar_fname
+                            + for_meta_bound_type
+                            + str(threshold))
+
+                meta = dict(bounds=bounds)
```

Comment:
Instead of creating strings and the `for_meta_bound_type` variable, let's simplify this by using the `self.wm_criterion` as the metadata dict. Or am I missing something?

Reply:
We previously outputted a filename and a threshold, so I was trying to preserve that. but I think outputting self.wm_criterion and a list of filenames might make the most sense.

Reply:
Yep - this looks good!